### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Krzychu92**
https://krzychu92.github.io/react_add-todo-form

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,61 +1,131 @@
...
+}
+
+export const tasks = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [taskUser, setTaskUser] = useState('0');
+  const [task, setTask] = useState('');
```
Comment (yevhenii-pyl) [inline src/App.tsx:22]: `task` is an object, right? 
You can't set a string as a default value to something that should contain an object. Default value should be empty object or, if you want to emphasise that it is empty, you may choose undefined.

File: src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,131 @@
...
+  const [errorTitle, setTitleError] = useState(false);
```
Comment (yevhenii-pyl) [inline src/App.tsx:23]: same here, 
you probably want to place a string here when there is an error, thus it should be either '' or undefined.

File: src/App.tsx [hunk 3]
```diff
@@ -1,61 +1,131 @@
...
+  const [errorUser, setUserError] = useState(false);
+  const [todos, setTasks] = useState(tasks);
```
Comment (yevhenii-pyl) [inline src/App.tsx:25]: tasks here a based on some API response that was additionally parsed and enhanced with user property. You can't use them as a default value. In theory, it's something async that will be added later. 
Default value should be just an empty array.

File: src/App.tsx [hunk 4]
```diff
@@ -1,61 +1,131 @@
...
+    setTitleError(false); // Reset error when user starts typing again
+  };
+
+  const handleUserChange = (event: ChangeEvent<HTMLSelectElement>) => {
+    setTaskUser(event.target.value);
+    setUserError(false); // Reset error when user starts typing again
+  };
+
+  const handleSubmit = (event: { preventDefault: () => void; }) => {
+    event.preventDefault();
```
Comment (yevhenii-pyl) [inline src/App.tsx:39]: Looks like you have already extracted preventDefault function from event.

File: src/components/TodoInfo/TodoInfo.tsx [hunk 5]
```diff
@@ -1 +1,30 @@
...
+import { Todo } from '../../Types/Todo';
+import './TodoInfo.scss';
+import usersFromServer from '../../api/users';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoInfo: React.FC<Props> = ({ todo }) => {
+  const {
```
Comment (yevhenii-pyl) [inline src/components/TodoInfo/TodoInfo.tsx:12]: Good job, you can and should extract properties here or at the highest level.

File: src/components/UserInfo/UserInfo.tsx [hunk 6]
```diff
@@ -1 +1,14 @@
-export const UserInfo = () => {};
+import { User } from '../../Types/User';
+import './UserInfo.scss';
+
+type Props = {
+  user: User;
+};
+
+export const UserInfo = ({ user }: Props) => {
```
Comment (yevhenii-pyl) [inline src/components/UserInfo/UserInfo.tsx:8]: Let's extract name and mail here.

**Review by yevhenii-pyl — CHANGES_REQUESTED** [review]
Good job, a few adjustments and it's good to go 👍 

As a general advice here: please, consider your variable naming. It may take some time but always worth it.

**Comment by Krzychu92** [comment]
hey, I have problem with some conflicts but I cannot fix it. May you help me ? I was trying to change some things what you commented but now I cant fix that with conflicts

**Review by Zibi95 — CHANGES_REQUESTED** [review]
Hey Krzysztof,
I noticed this is a really old PR. I assume the conflicts shouldn't be too hard to resolve. It seems that most of the changes during this time were related to the config files, including an upgrade to Node 20 and a change in the build tool. There's a QA session today at 6 PM, or you could simply create a new fork.

**Pull request closed without merging**